packages/block-editor/src/components/block-navigation/tree.js

```diff
@@ -17,6 +23,16 @@ import useBlockNavigationDropZone from './use-block-navigation-drop-zone';
 import { store as blockEditorStore } from '../../store';
 
 const noop = () => {};
+const expanded = ( state, action ) => {
```

Comment:
We only record state when someone toggles an expand/collapse chevron. 

One case I haven't covered is when blocks are deleted, we might have some extra state left over eg:

```jsx
state = { 
   'block-client-id-a': false,
   'block-client-id-b': true,
   'deleted-client-id': false
};
```

It is possible to prune this by dispatching an event as a useEffect side effect when `clientIdsTree`, or something similar like listening to a block delete event. I chose to leave this out since it does feel a bit unnecessary since this is likely to remain pretty small. Let me know if others feel otherwise and I can add handling.

Reply:
Yeah, agreed, it's probably not a big issue. I doubt this will ever become a huge amount of data.

I have ideas but I think it's an effort vs. reward question more than anything:
- BlockNavigation could receive stable block data from a selector (e.g. from `getBlockOrder`) that contains all clientIds, and reconcile it against the expanded state in an effect.
- It could be possible to move the state to the treegrid component entirely. Could each `TreeGridRow` manage its own state? I think it'd then have to render its own expander component, but that seems potentially ok. I might be missing some complications.